pylinalg/func/vector.py

```diff
@@ -116,3 +116,195 @@ def vector_apply_matrix(vectors, matrix, /, *, w=1, out=None, dtype=None):
     if dtype is not None:
         out = out.astype(dtype, copy=False)
     return out
+
+
+def vector_unproject(vector, projection_matrix, /, *, depth=0, out=None, dtype=None):
+    """
+    Un-project a vector from 2D space to 3D space.
+
+    Find a ``vectorB`` in 3D euclidean space such that the projection
+    ``projection_matrix @ vectorB`` yields the provided vector (in 2D euclidean
+    space). Since the solution to the above is a 1D subspace of 3D space (a
+    line), ``depth`` is used to select a single vector within.
+
+    Parameters
+    ----------
+    vector : ndarray, [2]
```

Comment:
I think this should be `[3]`?

Reply:
Actually, the way this is implemented right now in pygfx both `project` and `unproject` are functions that are applied to matrices, but that live in `vector3.py`:

```python
    def unproject(self, camera) -> "Vector3":
        return self.apply_matrix4(camera.projection_matrix_inverse).apply_matrix4(
            camera.matrix_world
        )
```

Should we match that signature in `unproject` and move the function around?

Reply:
The way this is implemented right now is that we take a 3-component vector in homogeneous space (u, v, 1) and project it into a 4-component vector in homogeneous space (x, y, z, 1).

@Korijn would you prefer to use homogeneous coordinates here, or would it be easier to have an interface that uses euclidean coordinates?

Reply:
Then I'd say we keep this as is (`array, [2]`).